### PR TITLE
Added support for MQTT LWT (status) messages

### PR DESCRIPTION
**Pull request by ghost**
This pull request adds support of birth and LWT (last will / status) messages for MQTT, also configuring them through Home Assistant MQTT Discovery. It lets Home Assistant know when A/C is not available for control.

### CODE REVIEW

**Comment by mamrai1** [comment]
I have tried your patch but after a while I see N/A values in HA climate...

**Comment by ghost** [comment]
> I have tried your patch but after a while I see N/A values in HA climate...

Hmm... these exact changes didn't cause such strange behaviour when they were applied on a bit older version (approx. May or June 2020). Could you please attach a screenshot of entity attributes and control card? Will investigate at weekend.

**Comment by mamrai1** [comment]
Yes, indeed... i have also reverted the code before #105 fixes and works as expected..it's been running on 2 units since yesterday and everything looks ok..

File: src/mitsubishi2mqtt/mitsubishi2mqtt.ino [hunk 1]
```diff
@@ -161,6 +161,7 @@ void setup() {
       ha_debug_topic           = mqtt_topic + "/" + mqtt_fn + "/debug";
       ha_debug_set_topic       = mqtt_topic + "/" + mqtt_fn + "/debug/set";
       ha_custom_packet         = mqtt_topic + "/" + mqtt_fn + "/custom/send";
+      ha_availability_topic = mqtt_topic + "/" + mqtt_fn + "/availability";
```
Comment (melyux) [inline src/mitsubishi2mqtt/mitsubishi2mqtt.ino:164]: Can you indent the equals sign to match the rest of the previous lines?

File: src/mitsubishi2mqtt/mitsubishi2mqtt.ino [hunk 2]
```diff
@@ -1503,6 +1504,9 @@ void haConfig() {
   haConfig["mode_stat_tpl"]                 = F("{{ value_json.mode if (value_json is defined and value_json.mode is defined and value_json.mode|length) else 'off' }}"); //Set default value for fix "Could not parse data for HA"
   haConfig["temp_cmd_t"]                    = ha_temp_set_topic;
   haConfig["temp_stat_t"]                   = ha_state_topic;
+  haConfig["avty_t"]                        = ha_availability_topic; // MQTT last will (status) messages topic 
+  haConfig["pl_not_avail"]                  = mqtt_payload_unavailable; // MQTT offline message payload
+  haConfig["pl_avail"]                      = mqtt_payload_available; // MQTT online message payload
```
Comment (melyux) [inline src/mitsubishi2mqtt/mitsubishi2mqtt.ino:1509]: Does the JsonDocument capacity set for this array earlier have to be increased to accommodate adding these three new fields?

**Review by melyux — CHANGES_REQUESTED** [review]

File: src/mitsubishi2mqtt/mitsubishi2mqtt.ino (hunk 2 above)
Comment (lesleypersyn) [reply src/mitsubishi2mqtt/mitsubishi2mqtt.ino:1509]: According to [https://arduinojson.org/v6/assistant/](https://arduinojson.org/v6/assistant/) this is the correct calculation: 
`const size_t capacity = 2*JSON_ARRAY_SIZE(6) + JSON_ARRAY_SIZE(7) + JSON_OBJECT_SIZE(5) + JSON_OBJECT_SIZE(29);`

Comment (mamrai1) [reply src/mitsubishi2mqtt/mitsubishi2mqtt.ino:1509]: No need. ..Checked and working as expected..It's on a different topic under mitsubishi2mqtt/xxxx/availability...

**Comment by gysmo38** [comment]
it still necessary to merge this change?

**Review by lesleypersyn — CHANGES_REQUESTED** [review]

**Comment by lesleypersyn** [comment]
> it still necessary to merge this change?

yes, please. Thanks!

**Comment by chrisallen** [comment]
Is there anything left before this can be merged? It would be nice for Home Assistant to know when A/C is not available for control.

**Comment by melyux** [comment]
@gysmo38

**Pull request merged**